src/SFA.DAS.Comitments.AcademicYearEndProcessor.WebJob/Updater/AcademicYearEndExpiryProcessor.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using SFA.DAS.Commitments.Domain.Data;
+using SFA.DAS.Commitments.Domain.Entities.DataLock;
+using SFA.DAS.Commitments.Domain.Interfaces;
+
+namespace SFA.DAS.Comitments.AcademicYearEndProcessor.WebJob.Updater
+{
+    public class AcademicYearEndExpiryProcessor : IAcademicYearEndExpiryProcessor
+    {
+        private readonly IAcademicYearDateProvider _academicYearProvider;
+        private readonly IDataLockRepository _dataLockRepository;
+        private readonly DataLockErrorCode _expirableErrorCodes;
+        private readonly ICurrentDateTime _currentDateTime;
+        private List<DataLockStatus> _expirableDatalocks = new List<DataLockStatus>();
+
+        public AcademicYearEndExpiryProcessor(IAcademicYearDateProvider academicYearProvider,
+            IDataLockRepository dataLockRepository, DataLockErrorCode expirableErrorCodes,
+            ICurrentDateTime currentDateTime)
+        {
+            _dataLockRepository = dataLockRepository ?? throw new ArgumentException("dataLockRepository");
+            _currentDateTime = currentDateTime ?? throw new ArgumentException("currentDateTime");
+            _academicYearProvider = academicYearProvider ?? throw new ArgumentException("academicYearProvider");
+            _expirableErrorCodes = expirableErrorCodes;
+        }
+
+        public async Task RunUpdate()
+        {
+
+            if (_currentDateTime.Now > _academicYearProvider.CurrentAcademicYearEndDate ||
+                _currentDateTime.Now < _academicYearProvider.CurrentAcademicYearStartDate)
+            {
+                throw new InvalidAcademicYearException($"The current time {_currentDateTime.Now} does not fall within the specified Academic Year: {_academicYearProvider.CurrentAcademicYearStartDate}-{_academicYearProvider.CurrentAcademicYearEndDate}.");
```

Comment:
Not completely convinced this is required - current academic year dates are calculated around "now", so this should never happen.

Reply:
We are currently relying on hard coded stuff that will be replaced by some sort of service outside of our direct control. If the academic year is wrong then its a big deal to the processing, potentially resulting in that data locks that should be expired are not. The reason for that should be obvious to support, that the dates are what is wrong, in order to fix and re-run.